src/UID2.Client/UID2Encryption.cs

```diff
@@ -228,16 +229,17 @@ private static bool DoesTokenHaveValidLifetime(ClientType clientType, KeyContain
                     return true;
             }
 
-            return DoesTokenHaveValidLifetimeImpl(established, expiry, now, maxLifetimeSeconds, keys.AllowClockSkewSeconds);
+            //generatedOrNow allows "now" for token v2, since v2 does not contain a "token generated" field. v2 therefore checks against remaining lifetime rather than total lifetime.
+            return DoesTokenHaveValidLifetimeImpl(generatedOrNow, expiry, now, maxLifetimeSeconds, keys.AllowClockSkewSeconds);
         }
 
 
-        private static bool DoesTokenHaveValidLifetimeImpl(DateTime established, DateTime expiry, DateTime now, long maxLifetimeSeconds, long allowClockSkewSeconds)
+        private static bool DoesTokenHaveValidLifetimeImpl(DateTime generatedOrNow, DateTime expiry, DateTime now, long maxLifetimeSeconds, long allowClockSkewSeconds)
```

Comment:
I would maybe change it to `DateTime? generated` since we already have a `now` param.

Reply:
or rename `now` param to `nowForClockSkewReference` to more distinctly differentiate from `generatedOrNow`

Reply:
There's only 1 caller, and we'd need null coalescing if we made it nullable, so will leave this for now